src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs

```diff
@@ -207,9 +207,26 @@ internal sealed partial class ExpressionEval {
             return instance.Call(p.Name, ArgumentSet.Empty);
         }
 
-        private IMember TryEvaluateWithArguments(IPythonModule module, FunctionDefinition fd, IArgumentSet args) {
+
+        private readonly Dictionary<int, IMember> _argEvalCache = new Dictionary<int, IMember>();
+
+        private IMember TryEvaluateWithArguments(IPythonFunctionType fn, IArgumentSet args) {
+            var name = fn.DeclaringType != null ? $"{fn.DeclaringModule.Name}.{fn.Name}" : fn.Name;
```

Comment:
This might be affected by #979 where the DeclaringModule of a function I had was `null`, so all functions of that style with the same name would get cached together.

Reply:
Yes, but it is a bug. I'll work on it right after this. All types should have declaring module, there are many many places where we would crash otherwise.